vendor/github.com/docker/libnetwork/cluster/provider.go

```diff
@@ -14,7 +28,7 @@ type Provider interface {
 	GetAdvertiseAddress() string
 	GetDataPathAddress() string
 	GetRemoteAddressList() []string
-	ListenClusterEvents() <-chan struct{}
+	ClusterEventsChannel() chan ConfigEventType
```

Comment:
Maybe just `ClusterEvents`? The data type usually isn't included in a function name - for example Go's `context` package has a `Done` method instead of `DoneChannel`.